tests/test_integration.py

```diff
@@ -1,1048 +1,1069 @@
-        self.assertEqual(surf.periodic_surface, problem.surfaces[3])
-        del surf.periodic_surface
-        self.assertIsNone(surf.periodic_surface)
-        with self.assertRaises(TypeError):
-            surf.periodic_surface = 5
-
-    def test_surface_transform(self):
-        problem = montepy.read_input("tests/inputs/test_surfaces.imcnp")
-        surf = problem.surfaces[1]
-        transform = problem.data_inputs[0]
-        del surf.periodic_surface
-        surf.transform = transform
-        self.assertEqual(surf.transform, transform)
-        self.assertIn("1 1 SO", surf.format_for_mcnp_input((6, 2, 0))[0])
-        del surf.transform
-        self.assertIsNone(surf.transform)
-        self.assertIn("1 SO", surf.format_for_mcnp_input((6, 2, 0))[0])
-        with self.assertRaises(TypeError):
-            surf.transform = 5
-
-    def test_materials_setter(self):
-        problem = copy.deepcopy(self.simple_problem)
-        with self.assertRaises(TypeError):
-            problem.materials = 5
-        with self.assertRaises(TypeError):
-            problem.materials = [5]
-        size = len(problem.materials)
-        problem.materials = list(problem.materials)
-        self.assertEqual(len(problem.materials), size)
-        problem.materials = problem.materials
-        self.assertEqual(len(problem.materials), size)
-
-    def test_reverse_pointers(self):
-        problem = self.simple_problem
-        complements = list(problem.cells[99].cells_complementing_this)
-        self.assertIn(problem.cells[5], complements)
-        self.assertEqual(len(complements), 1)
-        cells = list(problem.materials[1].cells)
-        self.assertIn(problem.cells[1], cells)
-        self.assertEqual(len(cells), 1)
-        cells = list(problem.surfaces[1005].cells)
-        self.assertIn(problem.cells[2], problem.surfaces[1005].cells)
-        self.assertEqual(len(cells), 2)
-
-    def test_surface_card_pass_through(self):
-        problem = montepy.read_input("tests/inputs/test_surfaces.imcnp")
-        surf = problem.surfaces[1]
-        # Test input pass through
-        answer = ["1 -2 SO -5"]
-        self.assertEqual(surf.format_for_mcnp_input((6, 2, 0)), answer)
-        # Test changing periodic surface
-        new_prob = copy.deepcopy(problem)
-        surf = new_prob.surfaces[1]
-        new_prob.surfaces[2].number = 5
-        self.assertEqual(int(surf.format_for_mcnp_input((6, 2, 0))[0].split()[1]), -5)
-        # Test changing transform
-        new_prob = copy.deepcopy(problem)
-        surf = new_prob.surfaces[4]
-        surf.transform.number = 5
-        self.assertEqual(int(surf.format_for_mcnp_input((6, 2, 0))[0].split()[1]), 5)
-        # test changing surface constants
-        new_prob = copy.deepcopy(problem)
-        surf = new_prob.surfaces[4]
-        surf.location = 2.5
-        self.assertEqual(
-            float(surf.format_for_mcnp_input((6, 2, 0))[0].split()[-1]), 2.5
-        )
+@pytest.fixture
```

Comment:
You should make this have a scope so it's ran once. Not being able to cache these problems really slowed down the test suite for a while. This is what I came up with recently: https://github.com/idaholab/MontePy/pull/488/files#diff-4071f2914f1a922463edf531fe1ccb42d1f0534ecb7f22b7812d955c840f1f46R295-R304

Reply:
Set the scope with [5460cb2](https://github.com/idaholab/MontePy/pull/493/commits/5460cb22d0ba8fcc63a88c5e667a223327196671)

After all the tests have been converted to pytest, we should consider making a global fixtures module. `test.imcnp` gets used a lot, for instance.